### PR TITLE
feat(codeowners): Manage external users without access to project

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective:
External Users that are not apart of teams that are a part of the project will fail CODEOWNERS sync and upload. We should handle similar to External Teams and show in the warning. This should not fail the upload/sync.

## UI:
codeowners-alert-users-without-access

## Tests:
Added integration test.

### CODE REVIEW

**Review by leeandher — COMMENTED** [review]
looks good, just a few questions

File: static/app/views/settings/project/projectOwnership/index.tsx [hunk 1]
```diff
@@ -210,9 +210,18 @@ tags.sku_class:enterprise #enterprise`;
               return errMessageListComponent(
                 `The following team do not have access to the project: ${project.slug}`,
                 values,
-                value => `/settings/${organization.slug}/teams/${value}/projects/`,
+                value =>
+                  `/settings/${organization.slug}/teams/${value.slice(1)}/projects/`,
```
[the remark this answers is not in the document]
Comment (NisanthanNanthakumar) [reply static/app/views/settings/project/projectOwnership/index.tsx:214]: @leeandher the type for `value` is `string`. So we should be able to use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice

File: static/app/views/settings/project/projectOwnership/index.tsx [hunk 2]
```diff
@@ -210,9 +210,18 @@ tags.sku_class:enterprise #enterprise`;
...
                 value => `Configure ${value} Permissions`
               );
+
+            case 'users_without_access':
+              return errMessageListComponent(
+                `The following users are not on a team that has access to the project: ${project.slug}`,
+                values,
+                _ => `/settings/${organization.slug}/members/`,
```
[the remark this answers is not in the document]
Comment (NisanthanNanthakumar) [reply static/app/views/settings/project/projectOwnership/index.tsx]: @leeandher this should be fixed. I added a query param to filter list of members for the email of the member with the missing project.

File: static/app/views/settings/project/projectOwnership/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (leeandher) [reply static/app/views/settings/project/projectOwnership/index.tsx:214]: Ohhhh my mistake, it just trims the '#'. You're totally right

**Review by leeandher — APPROVED** [review]
Looks good, it'd be nice if the query links for the users populated in the search bar instead of just showing the filtered results but if there isnt some easy way to do that, its obviously out of scope.


image

**Pull request merged**